src/Symfony/Component/Security/Core/Tests/Authentication/Token/AbstractTokenTest.php

```diff
@@ -96,6 +97,19 @@ public function testSerialize()
         $this->assertEquals($token->getAttributes(), $uToken->getAttributes());
     }
 
+    public function testSerializeWithRoleObjects()
```

Comment:
Your change is covered by the test above now.

> The issue here was when the roles are the same reference as the ones in the user entity.

Im not sure how that relates to your change... what about adding a `$token->setUser($mockedUser)` to above test so `serialize()` is fully covered.